pkg/controllers/update.go

```diff
@@ -149,3 +243,127 @@ func CLIChangeLog() (map[string]models.ChangeLog, http.Error) {
 	changes := models.ParseChangeLog(response)
 	return changes, http.Error{}
 }
+
+func InstallUpdate(version string) {
+	utils.Print("Updating...")
+
+	var wasUpdated bool
+	var installedVersion string
+	var controllerErr Error
+	if utils.IsWindows() && !utils.IsMINGW64() {
+		if InstalledViaWinget() {
+			wasUpdated, installedVersion, controllerErr = UpdateViaWinget(version)
+		} else {
+			utils.HandleError(fmt.Errorf("updates are not supported when installed via scoop. Please install the Doppler CLI via winget or update manually via `scoop update doppler`"))
+		}
+	} else {
+		wasUpdated, installedVersion, controllerErr = RunInstallScript()
+	}
+	if !controllerErr.IsNil() {
+		utils.HandleError(controllerErr.Unwrap(), controllerErr.Message)
+	}
+
+	if wasUpdated {
+		utils.Print(fmt.Sprintf("Installed CLI %s", installedVersion))
+
+		if changes, apiError := CLIChangeLog(); apiError.IsNil() {
+			utils.Print("\nWhat's new:")
+			printer.ChangeLog(changes, 1, false)
+			utils.Print("\nTip: run 'doppler changelog' to see all latest changes")
+		}
+
+		utils.Print("")
+	} else {
+		utils.Print("You are already running the latest version")
+	}
+
+	versionCheck := models.VersionCheck{LatestVersion: installedVersion, CheckedAt: time.Now()}
+	configuration.SetVersionCheck(versionCheck)
+}
+
+func InstalledViaWinget() bool {
+	utils.LogDebug("Checking if CLI is installed via winget")
+	command := fmt.Sprintf("winget list --id %s -n 1 --exact --disable-interactivity", wingetPackageId)
+	utils.LogDebug(fmt.Sprintf("Executing \"%s\"", command))
+
+	var out io.Writer = nil
+	if utils.CanLogDebug() {
+		out = os.Stderr
+	}
+	cmd, err := utils.RunCommandString(command, os.Environ(), nil, out, out, true)
+	if err != nil {
+		utils.LogDebugError(err)
+	}
+	_, err = utils.WaitCommand(cmd)
+	if err != nil {
```

Comment:
check: If `RunCommandString` fails bit `WaitCommand` succeeds, won't it overwrite `err` is `nil`? If so, we'd correctly print the error but we'd return the incorrect bool from this function.

Other calls to `RunCommandString` appear to return out of their error handler

Reply:
Whoops, this is missing a `return false`. Good catch.

The return value technically isn't accurate since an error here doesn't necessarily mean that the CLI isn't installed via winget, but I don't have a better pattern for now. I think it's also unlikely that we'll hit this.